src/lib.rs

```diff
@@ -102,12 +103,51 @@ pub fn select_coin_lowestlarger(
 /// Return None, if no solution exists.
 pub fn select_coin_fifo(
     inputs: Vec<OutputGroup>,
-    opitons: CoinSelectionOpt,
+    options: CoinSelectionOpt,
     excess_strategy: ExcessStrategy,
 ) -> Result<Option<(Vec<u32>, WasteMetric)>, SelectionError> {
-    unimplemented!()
+    let mut totalvalue: u64 = 0;
+    let mut totalweight: u32 = 0;
+    let mut selectedinputs: Vec<u32> = Vec::new();
+    impl Ord for OutputGroup {
+        fn cmp(&self, other:&Self) -> Ordering{
+            self.creation_sequence.cmp(&other.creation_sequence)
+        }
+    }
+    impl PartialOrd for OutputGroup{
+        fn partial_cmp(&self, other:&Self) -> Option<Ordering>{
+            Some(self.cmp(other))
+        }
+    }
+    let mut sortedinputs = inputs.clone();
+    sortedinputs.sort_by(|a,b| a.creation_sequence.cmp(&b.creation_sequence));
+    for (index,input) in sortedinputs.iter().enumerate(){
+        if totalvalue >= options.target_value {
```

Comment:
If and only if the total value of the inputs are greater than the target value + the estimated_fee, we can have a valid coin selection. Else appropriate error has to be returned